internal/types/path.go

```diff
@@ -17,17 +17,87 @@ package types
 import (
 	"fmt"
 	"strconv"
+	"strings"
 )
 
+// Path represents field path type. It should be used wherever we work with paths or dot notation.
+// Path should be stored and passed as value. All of its methods return new values, not modifying the receiver's state.
+type Path struct {
+	s []string
+}
+
+// NewPath returns Path from a strings slice.
+func NewPath(path []string) Path {
+	p := Path{s: make([]string, len(path))}
+	copy(p.s, path)
+	return p
+}
+
+// NewPathFromString returns Path from path string. Path string should contain fields separated with '.'.
+func NewPathFromString(s string) Path {
+	path := strings.Split(s, ".")
+
+	return NewPath(path)
+}
+
+// String returns dot-separated path value.
+func (p Path) String() string {
+	return strings.Join(p.s, ".")
+}
+
+// Len returns path length.
+func (p Path) Len() int {
+	return len(p.s)
+}
+
+// Slice returns path values array.
+func (p Path) Slice() []string {
+	path := make([]string, len(p.s))
+	copy(path, p.s)
+	return path
+}
+
+// Suffix returns the last path element.
+func (p Path) Suffix() string {
+	if len(p.s) == 0 {
+		return ""
+	}
+	return p.s[p.Len()-1]
+}
+
+// Prefix returns the first path element.
+func (p Path) Prefix() string {
+	if len(p.s) == 0 {
+		return ""
+	}
+	return p.s[0]
+}
+
+// TrimSuffix returns a path without the last element.
+func (p Path) TrimSuffix() Path {
+	if len(p.s) == 0 {
+		return p
+	}
+	return NewPath(p.s[:p.Len()-1])
+}
+
+// TrimPrefix returns a copy of path without the first element.
+func (p Path) TrimPrefix() Path {
+	if len(p.s) == 0 {
```

Comment:
It is not expected, if I wanted to trim, I expect there will be at least 2 elements. 
My suggestion is panic.